insights/src/main/kotlin/com/algolia/instantsearch/insights/Insights.kt

```diff
@@ -0,0 +1,77 @@
+package com.algolia.instantsearch.insights
+
+import android.content.Context
+import androidx.work.Constraints
+import androidx.work.NetworkType
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.PeriodicWorkRequestBuilder
+import java.util.concurrent.TimeUnit
+
+
+class Insights internal constructor(
+    context: Context,
+    private val credentials: Credentials
+) {
+
+    private val preferences = context.sharedPreferences(credentials.indexName)
+
+    init {
+        PeriodicWorkRequestBuilder<WorkerEvent>(10, TimeUnit.MINUTES).also {
```

Comment:
Same thing, this should be customisable. 10 minutes is way too long. Average app use is around 2-4 minutes